### PR TITLE
doc: Add SIG info in readme

### LINKED ISSUE #4283: Update README with SIG list
Crossplane have special interest groups for different area, some groups are temporary and some are long term.
This list is not discoverable

The list is a combination of Slack channels and calendar meetings

Reminder about SIGs (Special Interest Groups): improving feedback cycles on the various changes and features we have in-flight. We decided to try creating a SIG channel for each feature. The owner of each feature is also encouraged to setup a regular call for reviewers and interested parties to give feedback and drive alignment. Here are the SIG channels for the features that are in active development:
[#sig-composition-environments](https://crossplane.slack.com/archives/C05BP6QFLUW)
[#sig-composition-functions](https://crossplane.slack.com/archives/C031Y29CSAE)
[#sig-e2e-testing](https://crossplane.slack.com/archives/C05C8CCTVNV)
[#sig-observe-only](https://crossplane.slack.com/archives/C04D5988QEA)
[#sig-provider-families](https://crossplane.slack.com/archives/C056YAQRV16)
[#sig-secret-stores](https://crossplane.slack.com/archives/C05BY7DKFV2)
[#sig-deletion-ordering](https://crossplane.slack.com/archives/C05BP8W5ALW)
[#sig-upjet-provider-resource-utilization](https://crossplane.slack.com/archives/C04QLETDJGN)

### PR DESCRIPTION
**Pull request by csantanapr**
### Description of your changes

Add SIG list in readme

Fixes #4283 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Added or updated unit **and** E2E tests for my change.
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,6 +64,18 @@ with the broader community is encouraged to join.
 * [Past meeting recordings]
 * [Community Calendar][community calendar]
 
+### Special Interest Groups (SIG)
+Each SIG collaborates in Slack and some groups have regular meetings, you can find the meetings in the [Community Calendar][community calendar].
```
Comment (negz) [inline README.md:68]: Could you wrap at 80 chars to match the rest of of the doc, please.

**Review by negz — APPROVED** [review]
Thanks @csantanapr! This just needs to be wrapped and then I think it looks good.

In future it would be nice to find a way to list what SIGs exist that doesn't require README updates. I'd like to be able to spin them up and down a little dynamically perhaps. This works for now though.

File: README.md (hunk 1 above)
Comment (negz) [reply README.md:68]: Ping. Waiting for this before merging. 😄

Comment (csantanapr) [reply README.md:68]: sure @negz 👍

**Comment by csantanapr** [comment]
@negz 
>In future it would be nice to find a way to list what SIGs exist that doesn't require README updates. I'd like to be able to spin them up and down a little dynamically perhaps. 

Open to suggestions that would be viable for the crossplane community, they can be part of the docs that get read from a file

File: README.md (hunk 1 above)
Comment (csantanapr) [reply README.md]: sorry that I didn't see the notification for this @negz 
I made the changes

**Review by phisco — APPROVED** [review]

**Pull request merged**